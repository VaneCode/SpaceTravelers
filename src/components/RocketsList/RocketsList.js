// Import
import { useSelector, shallowEqual, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { FetchedRockets } from '../../redux/rockets/rockets';
import 'bootstrap/dist/css/bootstrap.min.css';
import Rocket from '../Rocket/Rocket';

const RocketList = () => {
  // Get the list of books from the state
  const rocketsStatus = useSelector(
    (state) => state.rockets,
    shallowEqual,
  );
  // Dispatch the action to get the rockets from the API
  const dispatch = useDispatch();
  useEffect(() => {
    if (!rocketsStatus.rockets.length) {
      dispatch(FetchedRockets());
    }
  });
  return (
    <>
      <div className="d-flex flex-column" style={{ gap: '1rem' }}>
        {rocketsStatus.rockets.map((rocket) => (
          <Rocket
            key={rocket.id}
            rocket={rocket}
          />
        ))}
      </div>
    </>
  );
};

export default RocketList;
