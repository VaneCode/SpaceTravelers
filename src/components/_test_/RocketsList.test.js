import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import RocketsList from '../RocketsList/RocketsList';

describe('Rocket list test', () => {
  test('snapshot', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <RocketsList />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
