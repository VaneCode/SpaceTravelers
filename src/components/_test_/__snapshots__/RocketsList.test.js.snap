// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket list test snapshot 1`] = `
<div
  className="d-flex flex-column"
  style={
    Object {
      "gap": "1rem",
    }
  }
/>
`;
